andela/ah-scorpion-frontend
# #159993343 Enable Users to Create, View, Edit, Reply to and Delete Comments

#### What does this PR do?
- Enables users to view comments, like comments, edit comments, reply to comments, delete comments and view comment edit history
#### Description of Task to be completed?
- Added comment component for displaying comments
- Added comment history component to show comments history
- added Modal dialog for showing the history of comment history
#### How should this be manually tested?
- Use git to pull from the branch `ft-CRUD-comments-159993343`
- run `npm install` to install any new dependencies
- Ensure that your environment variables are up to date and run `source .env` to export them
- run `npm start -s` to start a server
- Login using any user
- on the dashboard, you should be able to select an article, if none is available, use the url `/read` to create one
- Open the article in readt mode and scroll to the bottom
- You should be able to:
  - Create a new comment
  - Reply to an existing comment
  - delete your own comment
  - view the edit history to your comment
  - Like or Dislike a comment

#### What are the relevant pivotal tracker stories?
- #159993343 (https://www.pivotaltracker.com/story/show/159993343)
#### Screenshots (if appropriate)
###### No user logged in
<img width="685" alt="image" src="https://user-images.githubusercontent.com/41117362/46070312-1b252e00-c186-11e8-93ed-291ebea39657.png">

###### Article with no comments
<img width="681" alt="image" src="https://user-images.githubusercontent.com/41117362/46070335-31cb8500-c186-11e8-89fb-71237a2bd056.png">

###### Posting a comment
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/41117362/46070380-558ecb00-c186-11e8-9009-9ba259f0c5cd.png">

###### Replying to  comment
<img width="682" alt="image" src="https://user-images.githubusercontent.com/41117362/46070402-64757d80-c186-11e8-842b-e914d6392ac3.png">

###### Updating comment
<img width="1680" alt="image" src="https://user-images.githubusercontent.com/41117362/46070436-7a833e00-c186-11e8-9d25-59358cb73342.png">

###### Edit history
<img width="613" alt="image" src="https://user-images.githubusercontent.com/41117362/46070444-840ca600-c186-11e8-8428-7210a5cdd6cf.png">
